### PR TITLE
Utilize ParmEd to simplify GROMACS workflows and parameter conversion

### PR DESCRIPTION
**Pull request by davidlmobley**
This swaps the interior of several functions on the `gromacs` side with equivalent functionality (same API) utilizing `ParmEd`. Additionally, import statements are modified for the new layout of `parmed` which now is invoked via `import parmed` rather than `import chemistry`. Additionally, a new `amber_to_gromacs` conversion functionality is implemented using ParmEd. The ParmEd slicing and splitting functionality makes various operations much easier.

Additional discussion on this is available here: https://github.com/MobleyLab/openmoltools/pull/7

Still to do: implement an energy evaluation test to validate AMBER-to-GROMACS conversion.

### CODE REVIEW

**Comment by davidlmobley** [comment]
It's worth noting that ParmEd makes acpype obsolete; for the time being the acpype functionality is retained, though I should probably deprecate it and it can be removed at a later date.

**Comment by kyleabeauchamp** [comment]
Nice work with tackling this, taking a look now.

File: openmoltools/utils.py [hunk 1]
```diff
@@ -479,3 +485,54 @@ def get_checkmol_descriptors( molecule_filename, executable_name = 'checkmol' ):
  
     return descriptors
 
+def amber_to_gromacs( molecule_name, in_prmtop, in_crd, out_top = None, out_gro = None): 
+    """Use ParmEd to convert AMBER prmtop and crd files to GROMACS format.
+
+    Requires
+    --------
+    Currently requires gromacs_support branch of ParmEd
```
Comment (kyleabeauchamp) [inline openmoltools/utils.py]: The gromacs_support branch has been merged into Parmed, right?  So this requirement can be changed to a version number requirement?

File: openmoltools/utils.py [hunk 2]
```diff
@@ -479,3 +485,54 @@ def get_checkmol_descriptors( molecule_filename, executable_name = 'checkmol' ):
...
+    Returns
+    -------
+    out_top : str
+        GROMACS topology file produced by ParmEd
+    out_gro : str
+        GROMACS coordinate file produced by ParmEd
+
+    Notes
+    -----
+        molecule_name is not currently used except to generate output file names if gro/top file names are not provided. It is an argument partly for API consistency.
```
Comment (kyleabeauchamp) [inline openmoltools/utils.py]: PS: I'm quite flexible if you find the `molecule_name` and `out_top` living as both inputs and ouputs confusing.  I think I've been sticking with this convention for legacy reasons, but I think streamlining your workflow is a compelling reason for cleaning up these legacy issues.  Obviously that would be a separate issue / pull request.

**Comment by kyleabeauchamp** [comment]
OK, things look reasonable to me.  However, I'm not a gromacs expert and I also haven't done very much with the parmed gromacs stuff, so YMMV.

**Comment by davidlmobley** [comment]
@kyleabeauchamp - I've fixed the version requirement, sort of (except it's overly complicated at the moment - will fix again when there is a better solution). In terms of `molecule_name` - I actually like this reasonably well as in some cases I can see having `molecule_name` would simplify the workflow. So I don't necessarily want to change at this point.

**Comment by kyleabeauchamp** [comment]
Sounds good

**Comment by davidlmobley** [comment]
Added `pytables` as a dependency for travis to address this issue: https://github.com/mdtraj/mdtraj/issues/852#issuecomment-110184684

**Comment by rmcgibbo** [comment]
I cut a new MDTraj release that fixes the `tables` bug -- the fixed binaries should be up on binstar in the next hour or so.

**Comment by davidlmobley** [comment]
I also just determined that ParmEd now requires OpenMM 6.3 which is not released yet, so travis will presumably fail until either it is released or I switch to installing `openmm-dev` temporarily. Probably that means I won't merge this until it is released, which I am told will hopefully be in the next couple weeks.

**Comment by kyleabeauchamp** [comment]
I'm fine either way, a temporary Travis downtime is ok

**Comment by davidlmobley** [comment]
Added energy cross-comparison to the test for format conversion. Probably can merge once I decide what to do about travis.

**Comment by davidlmobley** [comment]
I will probably merge once ParmEd updates its conda channel to a more recent version with the current package organization (i.e., needs `import parmed` rather than `import chemistry`). This is currently causing problems in the Travis tests.

There is still the issue that ParmEd needs OpenMM later than 6.2 and 6.3 is not out yet, but since `conda install openmm-dev` works, I am OK with temporarily (partially) broken travis tests until the release of 6.3.

**Comment by davidlmobley** [comment]
Travis-CI is unhappy because of this issue, apparently: https://github.com/ParmEd/ParmEd/issues/181 - the `parmed.py` script in the `parmed` package is not playing nice with Travis-CI. I'll need to ask @swails to make his 2.0beta2 release and update the omnia package. Raising issue on ParmEd momentarily.

**Comment by davidlmobley** [comment]
I *think* Travis-CI will be happy now, as a new ParmEd binary was just released. We'll see. I'm going to have to merge very soon (i.e. tonight) one way or the other as the current main branch is incompatible with the current releases of ParmEd (since `import chemistry` in `utils.py` is no longer valid), breaking stuff my people are doing.

Presumably we could work around that by specifying that an older version of ParmEd is required, though that would be silly since the fix is already in this PR. 

I just updated `setup.py` to change the version tag to 0.6.8 in preparation for doing an 0.6.8 release once this is merged (hopefully shortly).

**Comment by davidlmobley** [comment]
Python 2 passed as far as expected (up to the issue involving the requirement for OpenMM 6.3) but I had an issue with Python 3. Fixed now, hopefully.

**Comment by davidlmobley** [comment]
OK, that seems to do it. I'm merging - getting Travis to be completely happy will have to wait for OpenMM 6.3.

**Pull request merged**